### ISSUE TITLE
Using Regex and pattern/template doesn't work

### ISSUE THREAD

**Issue by micscopau** [Stuk/eslint-plugin-header#40]
**Issue:**
When using the regex pattern/template format like:
```
"header/header": [2, "block", [
        {"pattern": " Copyright \\d{4}", "template": " Copyright 2021"}, 
        "My Company"
    ]]
```
Each file is reporting as incorrect header. Using the original method of pointing to a ```.txt``` file works correctly, so I know plugin is functioning. I've tried every combination of not providing template, or doing ```"line"``` instead of block and none of them work when providing a template.

This is with eslint 7.6 and 7.25 (updating eslint did not fix the issue)

**Comment by micscopau** [comment]
Issue was having parenthesis in the header, and not escaping them.
Issue was having ```Copyright (C) 2021```, solution was to do ```Copyright \\(C\\) 2021```
